olp-cpp-sdk-dataservice-write/src/VersionedLayerClientImpl.cpp

```diff
@@ -390,57 +390,31 @@ VersionedLayerClientImpl::CancelBatch(const model::Publication& pub) {
 
 olp::client::CancellationToken VersionedLayerClientImpl::CancelBatch(
     const model::Publication& pub, CancelBatchCallback callback) {
-  std::string publicationId = pub.GetId().value_or("");
-  if (publicationId.empty()) {
-    callback(client::ApiError(client::ErrorCode::InvalidArgument,
-                              "Invalid publication", true));
-    return {};
-  }
+  const auto publication = pub;
+  const auto catalog = catalog_;
+  const auto settings = settings_;
 
-  auto self = shared_from_this();
-  auto cancel_context = std::make_shared<client::CancellationContext>();
-  auto id = tokenList_.GetNextId();
-  auto cancel_function = [=]() {
-    self->tokenList_.RemoveTask(id);
-    callback(client::ApiError(client::ErrorCode::Cancelled,
-                              "Operation cancelled.", true));
-  };
+  auto cancel_batch_task =
+      [catalog, publication,
+       settings](client::CancellationContext context) -> CancelBatchResponse {
+    const auto publication_id = publication.GetId().value_or("");
+    if (publication_id.empty()) {
+      return client::ApiError(client::ErrorCode::InvalidArgument,
+                              "Invalid publication", true);
```

Comment:
I would return the following message: "Invalid publication: publication ID missing"